File: java/src/main/java/javaprojet/Entity/MenuComplement.java
package javaprojet.Entity;

public class MenuComplement {
    private int id;
    private int menuId;
    private int complementId;
    private int quantite;

    public MenuComplement() {
    }

    public MenuComplement(int id, int menuId, int complementId, int quantite) {
        this.id = id;
        this.menuId = menuId;
        this.complementId = complementId;
        this.quantite = quantite;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getMenuId() {
        return menuId;
    }

    public void setMenuId(int menuId) {
        this.menuId = menuId;
    }

    public int getComplementId() {
        return complementId;
    }

    public void setComplementId(int complementId) {
        this.complementId = complementId;
    }

    public int getQuantite() {
        return quantite;
    }

    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }

    @Override
    public String toString() {
        return "MenuComplement{" +
                "id=" + id +
                ", menuId=" + menuId +
                ", complementId=" + complementId +
                ", quantite=" + quantite +
                '}';
    }
}

File: Symfony/BrasilBurgerGest/src/Entity/MenuComplement.php
<?php

namespace App\Entity;

use App\Repository\MenuComplementRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: MenuComplementRepository::class)]
class MenuComplement
{
    #[ORM\Id, ORM\GeneratedValue, ORM\Column(type:"bigint")]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity:Menu::class, inversedBy:"menuComplements")]
    #[ORM\JoinColumn(nullable:true)]
    private ?Menu $menu = null;

    #[ORM\ManyToOne(targetEntity:Complement::class, inversedBy:"menuComplements")]
    #[ORM\JoinColumn(nullable:true)]
    private ?Complement $complement = null;

    #[ORM\Column(type:"integer")]
    private ?int $quantite = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getMenu(): ?Menu
    {
        return $this->menu;
    }

    public function setMenu(?Menu $menu): static
    {
        $this->menu = $menu;

        return $this;
    }

    public function getComplement(): ?Complement
    {
        return $this->complement;
    }

    public function setComplement(?Complement $complement): static
    {
        $this->complement = $complement;

        return $this;
    }

    public function getQuantite(): ?int
    {
        return $this->quantite;
    }

    public function setQuantite(int $quantite): static
    {
        $this->quantite = $quantite;

        return $this;
    }
}
